### ISSUE TITLE
Coding styleguide and linting for Sass/CSS

### ISSUE THREAD

**Issue by FrancescoK** [eHealthAfrica/eHealthAfrica.github.io#21]
Suggestion:
- Use Sass in every project
- Adoption of the Sass guidelines by Hugo Giraudel (http://sass-guidelin.es/#too-long-didnt-read)
- Add [grunt-scss-lint](https://github.com/ahmednuaman/grunt-scss-lint) to Gruntfiles throughout and use the [suggested ruleset](http://sass-guidelin.es/#scss-lint) 
- Defer discussion of naming conventions and structure (e.g. OOCSS/BEM/etc.) to separate frontend/design topic
- Add best practices about inclusion of Bootstrap, the theme or similar vendor libraries

**Comment by tthew** [comment]
:+1:

**Comment by ally** [comment]
:+1: :facepunch:

**Comment by FrancescoK** [comment]
grunt-scss-lint has a Ruby dependency :-/

**Comment by tthew** [comment]
This might help: [sass-lint](https://github.com/sasstools/sass-lint) (needs to be evaluated)

**Comment by FrancescoK** [comment]
@tthew Evaluation done: https://github.com/sasstools/sass-lint/blob/master/bin/sass-lint.js

**Comment by FrancescoK** [comment]
Evaluation: https://github.com/sasstools/sass-lint/blob/master/lib/sass-lint.js

**Comment by tthew** [comment]
haha omg.

**Comment by tthew** [comment]
@ally @adamvert @FrancescoK.

So we had some conversations around this;  BEM seems to be the favoured model.  Do you think we could move forward on this?  Hugo's guidelines + BEM for new work?

**Comment by FrancescoK** [comment]
Yes, there seems to be a consensus around BEM. Personally I have always been reluctant because BEM is kinda ugly and seems overkill considering the relatively low complexity of our designs, but that’s not really a rational argument. Imho, let’s go forward with BEM. How to call things within BEM will be the more interesting discussion but hey :)
